ethstats/ethstats.go

```diff
@@ -551,16 +637,49 @@ func (s *Service) assembleBlockStats(block *types.Block) *blockStats {
 		} else {
 			header = s.les.BlockChain().CurrentHeader()
 		}
+		state, _ = s.les.BlockChain().State()
 		td = s.les.BlockChain().GetTd(header.Hash(), header.Number.Uint64())
 		txs = []txStats{}
 	}
 	// Assemble and return the block stats
-	author, _ := s.engine.Author(header)
+	author, _ := s.backend.Author(header)
 
 	// Add epoch info
 	epochSize := s.eth.Config().Istanbul.Epoch
 	blockRemain := epochSize - istanbul.GetNumberWithinEpoch(header.Number.Uint64(), epochSize)
 
+	// Add validator enode list
+	valsElectedWithoutEnode := s.backend.GetValidators(block.Number(), block.Hash())
+
+	valsElected := make([]validatorInfo, 0, len(valsElectedWithoutEnode))
+	for i := range valsElectedWithoutEnode {
+		valsElected = append(valsElected, validatorInfo{
+			Address:      valsElectedWithoutEnode[i].Address(),
+			BLSPublicKey: valsElectedWithoutEnode[i].BLSPublicKey(),
+			// NodePublicKey: s.backend.valEnodeTable.GetEnodeURLFromAddress(valsElectedWithoutEnode[i].Address()),
```

Comment:
is this not present yet?

Reply:
It is - but I think we're doing our best not to expose this (sentries, encrypted enode table, ...). So I figured I leave it out for now. WDYT?